src/cosalib/metal.py

```diff
@@ -0,0 +1,316 @@
+"""
+Base Class for building Metal-type disks.
+"""
+
+# Copyright 2018-2020 Red Hat, Inc
+# Licensed under the new-BSD license (http://www.opensource.org/licenses/bsd-license.php)
+
+import distutils.util as dutil
+import logging as log
+import os
+import os.path
+import sys
+import yaml
+
+cosa_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, f"{cosa_dir}/cosalib")
+sys.path.insert(0, cosa_dir)
+
+from cosalib.build import (
+    _Build,
+    BuildExistsError
+)
+from cosalib.ostree import _BuildOSTree
+from cosalib.cmdlib import (
+    get_basearch,
+    run_verbose,
+)
+
+
+# For padded, disk is actual plus 513M
+SIZE_PADDED = "513M"
+# For estimated sizes, inflate the actual rootfs by this percent
+SIZE_ESITMATED = 35
+# rootfs defaults to being 'xfs'
+DEFAULT_ROOTFS = "xfs"
+
+# VARIANTS are the type of disks that can build built
+# To define new variants add its definition below:
+VARIANTS = {
+    "qemu": {
+        "arch": "any",
+        "platform": "qemu",
+        "format": "qcow2",
+        "target_args": [
+            "-drive",
+            "if=virtio,id=target,format={format},file={outfile},cache=unsafe"
+        ],
+        "kargs": [
+            "console=tty0 console={terminal},115200n8",
+            "ignition.platform.id={platform}"
+        ],
+    },
+    "metal": {
+        "arch": "any",
+        "platform": "metal",
+        "format": "raw",
+        "kargs": [
+            "console=tty0 console={terminal},115200n8",
+            "ignition.platform.id={platform}"
+        ],
+        "target_args": [
+            "-drive",
+            "if=virtio,id=target,format={format},file={outfile},cache=unsafe"
+        ],
+    },
+    "dasd": {
+        "arch": "s390x",
+        "format":  "raw",
+        "platform": "metal",
+        "kargs": [
+            "ignition.platform.id={platform}"
```

Comment:
Minor: I think we can probably hardcode that one where it's used. All the images we produce should have an `ignition.platform.id` karg.